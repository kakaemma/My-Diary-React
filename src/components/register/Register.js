import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import registerUser from '../../actions/userActions';
import RegistrationForm from '../forms/RegistrationForm';

class Register extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      email: '',
      password: '',
    };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.registrationSuccessful === true) {
      nextProps.history.push('/');
    }
  }

  handleChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  handleSubmit = event => {
    event.preventDefault();
    const { name, email, password } = this.state;
    const { registerUser } = this.props;
    const userData = {
      name,
      email,
      password,
    };
    registerUser(userData);
  };

  render() {
    return (
      <div className="common">
        <div className="common-login-register">
          <div className="container">
            <RegistrationForm onchange={this.handleChange} handleSubmit={this.handleSubmit} />
          </div>
        </div>
      </div>
    );
  }
}
Register.propTypes = {
  registerUser: PropTypes.func.isRequired,
  registrationSuccessful: PropTypes.bool.isRequired,
  history: PropTypes.func.isRequired,
};
const mapStateToProps = state => ({
  registrationSuccessful: state.user.registrationSuccessful,
});
export { Register as RegisterTest };
export default connect(mapStateToProps, { registerUser })(Register);
