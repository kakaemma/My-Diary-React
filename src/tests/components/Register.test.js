import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { RegisterTest, mapStateToProps } from '../../components/register/Register';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
configure({ adapter: new Adapter() });

describe('Register', () => {
  let wrapper;
  const history = { push: jest.fn() };

  const props = {
    history,
    loading: false,
    handleSubmit: jest.fn(),
    handleChange: jest.fn(),
    registerUser: jest.fn(),
    registrationSuccessful: false,
  };
  const getEvent = (name = '', value = '') => ({
    preventDefault: jest.fn(),
    target: {
      name,
      value,
    },
  });
  beforeEach(() => {
    const store = mockStore({ intitialState: {} });
    wrapper = shallow(<RegisterTest {...props} store={store} />);
  });
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should not redirect if registration is not successful', () => {
    const nextProps = {
      registrationSuccessful: false,
    };
    wrapper.setProps({ ...nextProps });
    expect(history.push).not.toBeCalled();
  });
  it('should redirect if registration is successful', () => {
    const nextProps = {
      registrationSuccessful: true,
    };
    wrapper.setProps({ ...nextProps });
    expect(history.push).toBeCalled();
  });
  it('should set state when handleChange is called', () => {
    wrapper.instance().handleChange(getEvent('email', 'test@gmail.com'));
    expect(wrapper.state().email).toEqual('test@gmail.com');
  });
  it('should call registerUser when handleSubmit is called', () => {
    wrapper.instance().handleSubmit(getEvent());
    expect(props.registerUser).toBeCalled();
  });
  it('should mapSate to Props', () => {
    const mockState = {
      user: { registrationSuccessful: false },
    };
    const returnedValue = mapStateToProps(mockState);
    expect(returnedValue).toEqual({ registrationSuccessful: false });
  });
});
