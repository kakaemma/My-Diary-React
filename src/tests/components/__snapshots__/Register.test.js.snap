// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    registerUser={[MockFunction]}
    registrationSuccessful={false}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="common-login-register"
      >
        <div
          className="container"
        >
          <registrationForm
            handleSubmit={[Function]}
            onchange={[Function]}
          />
        </div>
      </div>,
      "className": "common",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="container"
        >
          <registrationForm
            handleSubmit={[Function]}
            onchange={[Function]}
          />
        </div>,
        "className": "common-login-register",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <registrationForm
            handleSubmit={[Function]}
            onchange={[Function]}
          />,
          "className": "container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSubmit": [Function],
            "onchange": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="common-login-register"
        >
          <div
            className="container"
          >
            <registrationForm
              handleSubmit={[Function]}
              onchange={[Function]}
            />
          </div>
        </div>,
        "className": "common",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <registrationForm
              handleSubmit={[Function]}
              onchange={[Function]}
            />
          </div>,
          "className": "common-login-register",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <registrationForm
              handleSubmit={[Function]}
              onchange={[Function]}
            />,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "handleSubmit": [Function],
              "onchange": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
